core/state_processor.go

```diff
@@ -549,6 +609,9 @@ func ProcessQiTx(tx *types.Transaction, chain ChainContext, updateState bool, cu
 		if utxo == nil {
 			return nil, nil, fmt.Errorf("tx %032x spends non-existent UTXO %032x:%d", tx.Hash(), txIn.PreviousOutPoint.TxHash, txIn.PreviousOutPoint.Index)
 		}
+		if utxo.Lock != nil && utxo.Lock.Cmp(currentHeader.Number(location.Context())) > 0 {
```

Comment:
have we enforced strictly greater than everywhere? i.e locked utxo can only spent after the lock number is passed and not on the lock number

Reply:
Yes, it is enforced strictly greater than everywhere